api/envoy/config/cluster/v3/cluster.proto

```diff
@@ -507,6 +507,42 @@ message Cluster {
       google.protobuf.UInt32Value hash_balance_factor = 2 [(validate.rules).uint32 = {gte: 100}];
     }
 
+    // Configuration for :ref:`slow start mode <arch_overview_load_balancing_slow_start>`.
+    message SlowStartConfig {
+      // Endpoint warming policy defines additional condition (apart from condition that endpoint creation duration should be within slow start window)
+      // for endpoint to enter slow start mode.
+      // Refer to :ref:`endpoint warming policy types <arch_overview_load_balancing_slow_start_endpoint_warming_policy_types>`
+      // for an explanation on each policy type.
+      enum EndpointWarmingPolicy {
+        // Endpoint would enter slow start immediately.
+        NO_WAIT = 0;
+
+        // Endpoint would enter slow start upon passing first active healthcheck.
```

Comment:
Say what happens if health checks are not configured?

Reply:
Should we keep mentioning "active" HC? As passive HC could also flip/ clear host health flag and it would be considered healthy and satisfy this endpoint warming policy.

Reply:
Per other comments, IMO we should remove this policy entirely from the API for now. I think the way it should work is:
1) If no active HC configured, enter slow start immediately
2) If active HC configured, enter slow start after the first transition to healthy

Reply:
Alright, i need to inspect the code on "how to check if active HC is configured per cluster".

Reply:
Looks like i could use criteria `host->healthChecker() != null` on any host (as hc is configured per cluster so either all of hosts have hc configured or none) in priority update callback.